src/base/static/components/molecules/tag-editor.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+import { translate } from "react-i18next";
+
+import { TextareaInput } from "../atoms/input";
+import TagName from "./tag-name";
+
+import mapseedApiClient from "../../client/mapseed-api-client";
+
+const TagContainer = styled("div")(props => ({
+  outline: "none",
+  boxSizing: "border-box",
+  width: "100%",
+  padding: "8px",
+  display: "flex",
+  alignItems: "center",
+  marginBottom: "8px",
+  borderTop: props.isSelected ? "1px solid #a8a8a8" : "1px dashed #aaa",
+  borderRight: props.isSelected ? "1px solid #c8c8c8" : "1px dashed #aaa",
+  borderBottom: props.isSelected ? "1px solid transparent" : "1px dashed #aaa",
+  borderLeft: props.isSelected ? "1px solid #c8c8c8" : "1px dashed #aaa",
+  borderRadius: "3px",
+  backgroundColor: props.isSelected
+    ? props.backgroundColor || "#6495ed"
+    : "transparent",
+
+  "&:hover": {
+    cursor: "pointer",
+    backgroundColor: props.backgroundColor || "#6495ed",
+    opacity: props.isSelected ? 1 : 0.3,
+  },
+}));
+
+const NoteBox = styled(TextareaInput)(props => ({
+  outline: "none !important",
+  padding: "0 8px 0 6px",
+  overflow: props.isFocused ? "visible" : "hidden",
+  height: props.isFocused ? "5rem" : "0.7rem",
+  lineHeight: props.isFocused ? "1rem" : "0.8rem",
+  background: "transparent",
+  borderLeft: "1px solid #fff !important",
+  borderRight: "none !important",
+  borderTop: "none !important",
+  borderBottom: "none !important",
+  color: "#fff",
+  fontStyle: "italic",
+  fontSize: "0.75rem",
+  fontWeight: "normal",
+  transition: "all .5s ease",
+}));
+
+class TagEditor extends Component {
+  state = {
+    isFocused: false,
+    note: this.props.placeTag ? this.props.placeTag.note : "",
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  render() {
+    const isSelected = !!this.props.placeTag;
+
+    return (
+      <TagContainer
+        isSelected={isSelected}
+        backgroundColor={this.props.backgroundColor}
+        onClick={async () => {
+          // Toggle the placeTag.
+          if (isSelected) {
+            await mapseedApiClient.placeTags.delete(this.props.placeTag.url);
+            this.props.onDeletePlaceTag(this.props.placeTag.id);
+          } else {
+            const tagData = await mapseedApiClient.placeTags.create(
+              this.props.placeUrl,
+              {
+                note: this.state.note,
+                place: this.props.placeUrl,
+                tag: this.props.tag.url,
+              },
+            );
+            this.props.onCreatePlaceTag(tagData);
+          }
+        }}
+      >
+        <TagName displayName={this.props.displayName} isSelected={isSelected} />
```

Comment:
Similar to my comment above. Instead of having a selector and passing in `displayName` as a prop, we should be able to replace this with the following:

```suggestion
        <TagName displayName={tag.displayName} isSelected={isSelected} />
```